service/storage_proxy.cc

```diff
@@ -4551,7 +4551,7 @@ class data_read_resolver : public abstract_read_resolver {
         auto mp = mutation_partition(s, m.partition());
         auto&& ranges = cmd.slice.row_ranges(s, m.key());
         bool always_return_static_content = cmd.slice.options.contains<query::partition_slice::option::always_return_static_content>();
-        mp.compact_for_query(s, m.decorated_key(), cmd.timestamp, ranges, always_return_static_content, is_reversed, limit);
+        mp.compact_for_query(s, m.decorated_key(), cmd.timestamp, ranges, always_return_static_content, false, limit);
         return primary_key{m.decorated_key(), get_last_reconciled_row(s, mp, is_reversed)};
```

Comment:
Does reconciliation actually work after this PR? If everything read-related is "native", isn't the reconciliation result going to be reversed, and shouldn't we be unreversing it somewhere before applying it to replicas?

What tests did you run for this PR?

Reply:
I run the following tests

1. Start a 2-node cluster, create a table with RF=2 and populate it with data
2. Stop one of the nodes and delete all its sstable files.
3. Restart the node
4. Execute a reverse read on either of the nodes with consistency level set to ALL
5. Execute a reverse/forward read with consistency level set to ONE on the node that was stopped

Then I altered my test case to not just delete all files but rather introduce range tombstones (also overlapping) by executing multiple `DELETE` statements on the running node. Similarly with `UPDATE` statements.

Both test cases were testing with paging turned on and off. Such a test is missing in dtest so it should end up there to validate reconciliation still works.

Reply:
Don't we have unit tests for reconciliation?